deep_reference_parser/prodigy/reference_to_token_annotations.py

```diff
@@ -177,19 +180,33 @@ def create_span(self, tokens, index, label):
             "label": label,
         }
 
+        if self.text:
+            span["text"] = token["text"]
+
         return span
 
     def split_long_span(self, tokens, span, start_label, end_label, inside_label):
         """
         Split a multi-token span into `n` spans of lengh `1`, where `n=len(tokens)`
         """
-
         spans = []
-        spans.append(self.create_span(tokens, span["token_start"], start_label))
-        spans.append(self.create_span(tokens, span["token_end"], end_label))
```

Comment:
ah so was this line the cause of the bug in the case where span_size = 0 then?

Reply:
yep precisely! It was a case that had never come up before because references were always longer than one token.